libs/utils/energy_model.py

```diff
@@ -814,8 +1024,7 @@ def sge_path(cpu, domain, group, field):
         sge_file_values = read_multiple_oneline_files(target, sge_globs)
 
         if not sge_file_values:
-            raise TargetError('Energy Model not exposed in sysfs. '
-                              'Check CONFIG_SCHED_DEBUG is enabled.')
+            return cls.from_simplifiedEM_target(target)
```

Comment:
Hmmm what about splitting this more cleanly ? I'm thinking about something like this:

``` python
os = target.os
version = target.kernel_version.parts

if os == 'android' and parts >= (4, 14):
    return from_target_v2(...)
else:
    return from_target_v1(...)
```

Reply:
If we do this, we can't test the integration branch with it - there isn't a link between version or OS which we can rely on. Maybe we should refactor to provide 'is_this_model_present' methods..

Reply:
Right, then maybe we can try to determine which EM is there by looking at the sysfs entries instead.

Reply:
What we want to avoid is silent errors when there is actually nothing valid to read there. As long as it can fail properly, it should be fine.
Another solution could be to add a target config entry but it's easier if we can just autodetect it from the target.